src/service_provider/mod.rs

```diff
@@ -478,9 +475,9 @@ impl ServiceProvider {
                 value: entity_id,
                 ..Issuer::default()
             }),
-            name_id_policy: Some(NameIdPolicy {
+            name_id_policy: self.name_id_format().map(|format| NameIdPolicy {
                 allow_create: Some(true),
-                format: self.name_id_format(),
+                format: Some(format),
```

Comment:
My understanding is that the `format` field is optional on `NameIDPolicy`, so I don't think we should force it to be set here. I'm fine with returning the proper unspecified format value though. That was an oversight.

Reply:
I've thought about this some more, and decided this is fine